src/PrestaShopBundle/Entity/Shipment.php

```diff
@@ -0,0 +1,260 @@
+        $this->updatedAt = new DateTime('now');
+        $this->products = new ArrayCollection();
+    }
+
+    public function getProducts(): Collection
+    {
+        return $this->products;
+    }
+
+    public function getUpdatedAt()
+    {
+        return $this->updatedAt;
+    }
+
+    public function getCreatedAt()
+    {
+        return $this->createdAt;
+    }
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function getOrderId(): int
+    {
+        return $this->orderId;
+    }
+
+    public function getCarrierId(): int
+    {
+        return $this->carrierId;
+    }
+
+    public function getDeliveryAddressId(): int
+    {
+        return $this->deliveryAddressId;
+    }
+
+    public function getShippingCostTaxExcluded(): float
+    {
+        return $this->shippingCostTaxExcluded;
+    }
+
+    public function getShippingCostTaxIncluded(): float
+    {
+        return $this->shippingCostTaxIncluded;
+    }
+
+    public function getPackedAt(): ?DateTime
+    {
+        return $this->packedAt;
+    }
+
+    public function getShippedAt(): ?DateTime
+    {
+        return $this->shippedAt;
+    }
+
+    public function getDeliveredAt(): ?DateTime
+    {
+        return $this->deliveredAt;
+    }
+
+    public function getTrakingNumber(): ?string
+    {
+        return $this->trakingNumber;
+    }
+
+    public function setId(int $id): self
+    {
+        $this->id = $id;
+
+        return $this;
+    }
+
+    public function setOrderId(int $orderId): self
+    {
+        $this->orderId = $orderId;
+
+        return $this;
+    }
+
+    public function setCarrierId(int $carrierId): self
+    {
+        $this->carrierId = $carrierId;
+
+        return $this;
+    }
+
+    public function setDeliveryAddressId(int $deliveryAddressId): self
+    {
+        $this->deliveryAddressId = $deliveryAddressId;
+
+        return $this;
+    }
+
+    public function setShippingCostTaxExcluded(float $shippingCostTaxExcluded): self
+    {
+        $this->shippingCostTaxExcluded = $shippingCostTaxExcluded;
+
+        return $this;
+    }
+
+    public function setShippingCostTaxIncluded(float $shippingCostTaxIncluded): self
+    {
+        $this->shippingCostTaxIncluded = $shippingCostTaxIncluded;
+
+        return $this;
+    }
+
+    public function setPackedAt(?DateTime $packedAt): self
+    {
+        $this->packedAt = $packedAt;
+
+        return $this;
+    }
+
+    public function setShippedAt(?DateTime $shippedAt): self
+    {
+        $this->shippedAt = $shippedAt;
+
+        return $this;
+    }
+
+    public function setDeliveredAt(?DateTime $deliveredAt): self
+    {
+        $this->deliveredAt = $deliveredAt;
+
+        return $this;
+    }
+
+    public function setTrakingNumber(?string $trakingNumber): self
+    {
+        $this->trakingNumber = $trakingNumber;
+
+        return $this;
+    }
+
+    public function setProducts(Collection $products): self
```

Comment:
Maybe not needed yet but consider having an `addProduct` in the future I also believe you need to loop through the passed product instance to call their `$product->setShipment($this);` (though I'm not 100% sure if it's a must-have or if doctrine can handle this association from this side only)